### DISCUSSION TITLE
Welcome to HVM Discussions!

### DISCUSSION THREAD

**Discussion by steinerkelvin** [HigherOrderCO/HVM1#23] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by Swoorup** [comment]
Can hvm in theory optimise simple recursive fibonacci such as this? A quick test, it still suffers perf penalty.

```haskell
(Fib 0) = (0)
(Fib 1) = (1)
(Fib n) = (+ (Fib (- n 1)) (Fib (- n 2)))
```
